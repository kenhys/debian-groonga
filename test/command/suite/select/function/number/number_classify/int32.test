plugin_register functions/number

table_create Prices TABLE_PAT_KEY Int32

load --table Prices
[
{"_key": -201},
{"_key": -200},
{"_key": -199},
{"_key": -101},
{"_key": -100},
{"_key": -99},
{"_key": -1},
{"_key": 0},
{"_key": 1},
{"_key": 99},
{"_key": 100},
{"_key": 101},
{"_key": 199},
{"_key": 200},
{"_key": 201}
]

select Prices \
  --sortby _id \
  --limit -1 \
  --output_columns '_key, number_classify(_key, 100)'
