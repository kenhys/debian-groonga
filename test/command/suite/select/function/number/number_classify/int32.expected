plugin_register functions/number
[[0,0.0,0.0],true]
table_create Prices TABLE_PAT_KEY Int32
[[0,0.0,0.0],true]
load --table Prices
[
{"_key": -201},
{"_key": -200},
{"_key": -199},
{"_key": -101},
{"_key": -100},
{"_key": -99},
{"_key": -1},
{"_key": 0},
{"_key": 1},
{"_key": 99},
{"_key": 100},
{"_key": 101},
{"_key": 199},
{"_key": 200},
{"_key": 201}
]
[[0,0.0,0.0],15]
select Prices   --sortby _id   --limit -1   --output_columns '_key, number_classify(_key, 100)'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        15
      ],
      [
        [
          "_key",
          "Int32"
        ],
        [
          "number_classify",
          null
        ]
      ],
      [
        -201,
        -300
      ],
      [
        -200,
        -200
      ],
      [
        -199,
        -200
      ],
      [
        -101,
        -200
      ],
      [
        -100,
        -100
      ],
      [
        -99,
        -100
      ],
      [
        -1,
        -100
      ],
      [
        0,
        0
      ],
      [
        1,
        0
      ],
      [
        99,
        0
      ],
      [
        100,
        100
      ],
      [
        101,
        100
      ],
      [
        199,
        100
      ],
      [
        200,
        200
      ],
      [
        201,
        200
      ]
    ]
  ]
]
